test/test_nn.py

```diff
@@ -2605,6 +2605,15 @@ def test_mse_loss_size_warning(self):
             self.assertEqual(len(w), 1)
             self.assertIn('Please ensure they have the same size.', str(w[0]))
 
+    def test_mse_loss_backward_promotion(self):
```

Comment:
@zeshengzong do you have a smaller repro? You can call torch.ops.aten.mse_loss_backward directly

My confusion is that our out variants usually do not undergo type promotion.